### ISSUE TITLE
Diffsort implementation using Tensorflow

### ISSUE THREAD

**Issue by jimth001** [Felix-Petersen/diffsort#4]
I really appreciate your great work and want to try diffsort in my research. We have to use diffsort in tensorflow for some reasons. So recently I'm trying to rewrite this project with tensorflow. I opened this issue to record the problems I encountered in implementing diffsort, hoping to get your help. In the future, I will open source the version I implemented. It would be even better if you can release an official version of TF implementation.

**Comment by jimth001** [comment]
The NormalCDF function is implemented in lines 13-20 of functional.py, and the backward function is explicitly defines the gradient of the forward. However, it appears that the gradient implemented in the backward function is the standard gradient for the forward function, and I have tested it with a few examples to confirm that. I would like to know if my judgment is incorrect. If my judgment is correct, why not simply define the forward function in PyTorch and let PyTorch automatically calculate the gradient?

**Comment by Felix-Petersen** [comment]
Thanks for your interest. 
You are right, the backward of NormalCDF is simply the vanilla gradient of the forward. The reason for implementing a custom backward was numerical stability because `erf` doesn't have a closed form and is internally approximated with a polynomial approximation, so to be on the safe side, I just implemented the backward via the PDF of the distribution which has an exact closed form.
